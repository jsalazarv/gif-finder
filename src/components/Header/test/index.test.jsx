import { render } from '@testing-library/react';
import { Header } from '../index';
import { FinderInput } from '../../../GifFinder/components/FinderInput/index.jsx';

jest.mock('../../../GifFinder/components/FinderInput');

describe('[Header]', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('Should render the [Header] component', () => {
    const { container } = render(<Header />);

    expect(container.firstChild).toHaveClass('header');
    expect(container).toHaveTextContent('GIF');
    expect(container).toHaveTextContent('INDER');
    expect(container).toMatchSnapshot();
  });

  test('Should render [FinderInput]', async () => {
    render(<Header />);

    expect(FinderInput).toHaveBeenCalledOnce();
  });
});
