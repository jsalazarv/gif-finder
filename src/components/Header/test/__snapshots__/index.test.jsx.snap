// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Header] Should render the [Header] component 1`] = `
<div>
  <header
    class="header"
  >
    <nav
      class="navbar__content"
    >
      <h1
        class="navbar__title"
      >
        <span
          class="animate-text"
        >
          GIF
        </span>
        INDER
      </h1>
    </nav>
  </header>
</div>
`;
